File: frontend/src/types/comment.ts
export interface CommentAuthorSummary {
  _id: string;
  username: string;
  display_name: string;
  avatar_url: string;
}

export interface Comment {
  _id: string;
  post_id: string;
  author_id: string;
  author?: CommentAuthorSummary;
  content: string;
  created_at: string;
  is_owner: boolean;
}

export interface CommentListResponse {
  comments: Comment[];
  total: number;
}

File: backend/app/schemas/comment.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field, field_validator

from app.schemas.common import PyObjectId


class CommentAuthorSummary(BaseModel):
    id: PyObjectId = Field(..., alias="_id")
    username: str
    display_name: str
    avatar_url: str = ""

    model_config = ConfigDict(
        populate_by_name=True,
        from_attributes=True,
    )


class CommentCreateRequest(BaseModel):
    content: str = Field(..., min_length=1, max_length=500, description="Comment text")

    @field_validator("content")
    @classmethod
    def validate_content(cls, v: str) -> str:
        clean = v.strip()
        if not clean:
            raise ValueError("Comment cannot be blank or contain only whitespace")
        return clean


class CommentResponse(BaseModel):
    id: PyObjectId = Field(..., alias="_id")
    post_id: PyObjectId
    author_id: PyObjectId
    author: CommentAuthorSummary | None = None
    content: str
    created_at: datetime
    is_owner: bool = False

    model_config = ConfigDict(
        populate_by_name=True,
        from_attributes=True,
    )


class CommentListResponse(BaseModel):
    comments: list[CommentResponse]
    total: int
